engine/src/main/scala/cromwell/engine/workflow/tokens/JobTokenDispenserActor.scala

```diff
@@ -62,21 +59,21 @@ class JobTokenDispenserActor(override val serviceRegistryActor: ActorRef,
   private val tokensLeasedMetricPath: NonEmptyList[String] = tokenDispenserMetricsActivityRates :+ "tokens_dispensed"
   private val tokensReturnedMetricPath: NonEmptyList[String] = tokenDispenserMetricsActivityRates :+ "tokens_returned"
 
-  final private val groupMetricsTimeout = Timeout(60.seconds)
-
   /**
     * Lazily created token queue. We only create a queue for a token type when we need it
     */
   var tokenQueues: Map[JobTokenType, TokenQueue] = Map.empty
   var currentTokenQueuePointer: Int = 0
   var tokenAssignments: Map[ActorRef, TokenLeaseRecord] = Map.empty
+  var quotaExhaustedGroups: List[String] = List.empty
 
   val instrumentationAction = () => {
     sendGaugeJob(tokenAllocatedDescription, tokenAssignments.size.toLong)
     sendGaugeJob(ExecutionStatus.QueuedInCromwell.toString, tokenQueues.values.map(_.size).sum.toLong)
   }
 
   lazy val effectiveLogInterval: Option[FiniteDuration] = logInterval.filterNot(_ == 0.seconds)
+  lazy val quotaExhaustedGroupsRefreshInterval: FiniteDuration = 1.minutes
```

Comment:
TOL: do we want to increase this interval to 5 mins or maybe more? The threshold for a group to be no longer in quota exhausted state is 15 mins. Or do you think it would be more beneficial to pull frequently since it's just querying the database?

> threshold (in minutes) after which a group in GROUP_METRICS_ENTRY table is no longer considered to be actively experiencing quota exhaustion
    threshold-minutes = 15

Reply:
We could increase it, but that would effectively make the time that we wait for groups to stop being quota exhausted an additional five minutes - seems better to refresh quickly and one database query per minute doesn't seem too bad.

Reply:
Sounds good 👍